File: Redistribute_Characters_to_Make_All_Strings_Equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        unordered_map<char, int> chars;

        for (const string& word : words) {
            for (char c : word) {
                chars[c] = 1 + chars[c];
            }
        }

        int divider = words.size();

        for (const auto& entry : chars) {
            if (entry.second % divider != 0) {
                return false;
            }
        }

        return true;        
    }
};

File: Redistribute_Characters_to_Make_All_Strings_Equal.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
var makeEqual = function(words) {
    const chars = {};

    for (const word of words) {
        for (let i = 0; i < word.length; i++) {
            chars[word[i]] = 1 + (chars[word[i]] || 0);
        }
    }

    const divider = words.length;

    for (const freq of Object.values(chars)) {
        if (freq % divider !== 0) {
            return false;
        }
    }

    return true;    
};